ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -735,15 +735,16 @@ libBinParserModules ghcFlavor = do
 -- the sdists rather than these. If we don't remove them, some
 -- ambiguity results. Cabal and stack are OK with that but bazel gets
 -- messed up.
-removeGeneratedIntermediateFiles :: IO ()
-removeGeneratedIntermediateFiles = do
+removeGeneratedIntermediateFiles :: GhcFlavor -> IO ()
+removeGeneratedIntermediateFiles ghcFlavor = do
     removeFile $ stage0GhcHeap </> "GHC/Exts/Heap/Utils.hs"
     removeFile $ stage0GhcHeap </> "GHC/Exts/Heap/InfoTableProf.hs"
     removeFile $ stage0GhcHeap </> "GHC/Exts/Heap/InfoTable.hs"
     removeFile $ stage0GhcHeap </> "GHC/Exts/Heap/InfoTable/Types.hs"
     removeFile $ stage0GhcHeap </> "GHC/Exts/Heap/Constants.hs"
     removeFile $ stage0Ghci </> "GHCi/FFI.hs"
-    removeFile $ stage0Ghci </> "GHCi/InfoTable.hs"
+    when (ghcFlavor /= GhcMaster) $
+      removeFile $ stage0Ghci </> "GHCi/InfoTable.hs"
```

Comment:
Why do we only remove `GHCi/InfoTable.hs` but not `GHCi/StaticPtrTable.hs`, which seems to have changed in the same way?

Reply:
It is not a "generated intermediate file" as `InfoTable.hs` is.

Reply:
I see. Thanks for the explanation.

Reply:
You bet.